client/my-sites/plans/current-plan/trials/trial-current-plan.tsx

```diff
@@ -0,0 +1,139 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import {
+	PLAN_BUSINESS_MONTHLY,
+	PLAN_ECOMMERCE_TRIAL_MONTHLY,
+	PLAN_WOOEXPRESS_MEDIUM_MONTHLY,
+} from '@automattic/calypso-products';
+import { Button } from '@automattic/components';
+import { useMediaQuery } from '@wordpress/compose';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useState } from 'react';
+import BodySectionCssClass from 'calypso/layout/body-section-css-class';
+import { getECommerceTrialCheckoutUrl } from 'calypso/lib/trials/get-ecommerce-trial-checkout-url';
+import { useSelector } from 'calypso/state';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import ECommerceTrialBanner from '../../ecommerce-trial/ecommerce-trial-banner';
+import EcommerceTrialIncluded from './ecommerce-trial-included';
+import EcommerceTrialNotIncluded from './ecommerce-trial-not-included';
+import MigrationTrialIncluded from './migration-trial-included';
+
+import './trial-current-plan.scss';
+
+const TrialCurrentPlan = () => {
+	const selectedSite = useSelector( ( state ) => getSelectedSite( state ) );
+
+	const translate = useTranslate();
+
+	const [ showAllTrialFeaturesInMobileView, setShowAllTrialFeaturesInMobileView ] =
+		useState( false );
+
+	const viewAllIncludedFeatures = () => {
+		setShowAllTrialFeaturesInMobileView( true );
+	};
+
+	const currentPlanSlug = selectedSite?.plan?.product_slug ?? '';
+	const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
+	// const isMigrationTrial = currentPlanSlug === PLAN_MIGRATION_TRIAL_MONTHLY;
+
+	const isMobile = useMediaQuery( '(max-width: 480px)' );
+	const displayAllIncluded = ! isMobile || showAllTrialFeaturesInMobileView;
+	const bodyClass = isEcommerceTrial
+		? [ 'is-ecommerce-trial-plan' ]
+		: [ 'is-migration-trial-plan' ];
+
+	const targetPlan = isEcommerceTrial ? PLAN_WOOEXPRESS_MEDIUM_MONTHLY : PLAN_BUSINESS_MONTHLY;
+	const trackEvent = isEcommerceTrial
+		? 'calypso_wooexpress_my_plan_cta'
+		: 'calypso_migration_my_plan_cta';
+
+	/**
+	 * Redirects to the checkout page with Plan on cart.
+	 *
+	 * @param ctaPosition - The position of the CTA that triggered the redirect.
+	 */
+	const goToCheckoutWithPlan = ( ctaPosition: string ) => {
```

Comment:
Currently if I can the upgrade plan button at the bottom of the page in a migration trial plan. It gives me an error of `Sorry, that plan is not a valid upgrade from your current plan.` on the checkout page.